core/domain/base_model_validators.py

```diff
@@ -90,6 +93,44 @@ def __init__(
         self.model_ids = model_ids
```

Comment:
Should we produce an error here if `model_class` is `UserSettingsModel`?

Reply:
I had added a note, but an exception would be better I guess. Done.